### PR TITLE
SQL-1751, SQL-1752: Implement changes necessary for Direct Query support

### PR DESCRIPTION
**Pull request by pmeredit**
also, protect the download center from non-master tags

Changes to yml files can be ignored, all they do is update the metadata to match the metadata changes

NOTES: 
* beta will not be release until this PR is accepted
* PR will not be merged until after Beta ends

### CODE REVIEW

File: core/src/type_info.rs [hunk 1]
```diff
@@ -7,7 +7,8 @@ use odbc_sys::Nullability;
 
 use lazy_static::lazy_static;
 
-const DATA_TYPES: [BsonTypeInfo; 21] = [
```
Comment (pmeredit) [inline core/src/type_info.rs:10]: Power BI needs VARCHAR to appear in the type info

File: evergreen.yml [hunk 2]
```diff
@@ -1319,6 +1319,9 @@ functions:
         working_dir: mongosql-odbc-driver/resources/download-center/
         script: |
           ${prepare_shell}
```
Comment (pmeredit) [inline evergreen.yml:1321]: This is the change protecting the download center

File: odbc/src/api/definitions.rs [hunk 3]
```diff
@@ -622,11 +622,15 @@ pub const SQL_CVT_GUID: u32 = 0x01000000;
 pub const MONGO_CAST_SUPPORT: u32 = SQL_CVT_CHAR
     | SQL_CVT_NUMERIC
     | SQL_CVT_INTEGER
```
Comment (pmeredit) [inline odbc/src/api/definitions.rs:624]: We needed to report that we support conversions to BIGINT, WCHAR, WVARCHAR, and WLONGVARCHAR

File: odbc/src/api/definitions.rs [hunk 4]
```diff
@@ -645,6 +650,42 @@ pub const SQL_FN_NUM_POWER: u32 = 0x00100000;
 pub const SQL_FN_NUM_RADIANS: u32 = 0x00200000;
 pub const SQL_FN_NUM_ROUND: u32 = 0x00400000;
```
Comment (pmeredit) [inline odbc/src/api/definitions.rs:652]: Added for completeness

File: odbc/src/api/functions.rs [hunk 5]
```diff
@@ -1922,6 +1922,17 @@ macro_rules! sql_get_info_helper {
...
+                      | SQL_OJ_ALL_COMPARISON_OPS;
+                    i16_len::set_output_fixed_data(
+                        &OJ_CAPABILITIES,
+                        info_value_ptr,
+                        string_length_ptr,
+                    )
+                }
                 InfoType::SQL_CATALOG_NAME_SEPARATOR => {
                     // The name separator used by MongoSQL is '.'.
                     i16_len::set_output_wstring_as_bytes(
```
Comment (pmeredit) [inline odbc/src/api/functions.rs:1938]: These are new functions that we either incorrectly reported that we did not support, or that we added support to mongosql during the DQ epic

File: evergreen.yml [hunk 6]
```diff
@@ -1319,6 +1319,9 @@ functions:
...
+          if [[ $(git branch --show-current) != 'master' ]]; then
```
Comment (nbagnard) [inline evergreen.yml:1322]: 👍

File: odbc/src/api/col_attr_describe_tests.rs [hunk 7]
```diff
@@ -424,6 +424,7 @@ mod unit {
                         numeric_attr_ptr,
                     )
                 );
+                dbg!(desc);
```
Comment (nbagnard) [inline odbc/src/api/col_attr_describe_tests.rs:427]: Remaining dbg! ;)

File: core/src/bson_type_info.rs [hunk 8]
```diff
@@ -86,14 +86,36 @@ impl BsonTypeInfo {
     pub const STRING: BsonTypeInfo = BsonTypeInfo {
         type_name: "string",
         sql_type: SqlDataType::EXT_W_VARCHAR,
+        non_concise_type: SqlDataType::EXT_W_VARCHAR,
+        searchable: SQL_SEARCHABLE,
+        is_case_sensitive: true,
+        fixed_prec_scale: false,
+        scale: None,
+        precision: Some(65535),
```
Comment (nbagnard) [inline core/src/bson_type_info.rs:94]: Let's define a constant for the max size.

**Review by nbagnard — COMMENTED** [review]
I reviewed only the max length change, but not the new functions/convert yet

File: core/src/bson_type_info.rs [hunk 9]
```diff
@@ -86,14 +86,36 @@ impl BsonTypeInfo {
...
+        octet_length: Some(65535),
+        fixed_bytes_length: Some(65535),
```
[the remark this answers is not in the document]
Comment (pmeredit) [reply core/src/bson_type_info.rs:96]: aren't they both the size in bytes?

File: evergreen.yml [hunk 10]
```diff
@@ -343,7 +343,7 @@ functions:
               MINOR_VERSION="0.1"
               VERSION_LABEL="0.1.0"
           else
-              MINOR_VERSION=$(echo "$RELEASE_VERSION" | sed 's|\([0-9]\+[.][0-9]\+\)[.][0-9]\+|\1|')
+              MINOR_VERSION=$(echo "$RELEASE_VERSION" | sed 's|\([0-9]\+[.][0-9]\+\)[.][0-9]\+.*|\1|')
```
[the remark this answers is not in the document]
Comment (pmeredit) [reply evergreen.yml:346]: That's odd. Let me check this out. I do want it to return 1.2, yes

File: core/src/bson_type_info.rs (hunk 9 above)
[the remark this answers is not in the document]
Comment (pmeredit) [reply core/src/bson_type_info.rs:96]: I think the answer here is that fixed_bytes_length should still be None. It's also bytes, but it's for fixed sized data only.

File: odbc/src/api/col_attr_describe_tests.rs (hunk 7 above)
Comment (pmeredit) [reply odbc/src/api/col_attr_describe_tests.rs:427]: No, that was is intentional :D It shows us where the test fails iff it fails.

**Review by nbagnard — CHANGES_REQUESTED** [review]
I noted a few minor changes that I'd like you to integrate and then we should be good to go 👍

File: odbc/src/api/definitions.rs [hunk 11]
```diff
@@ -694,6 +735,27 @@ pub const SQL_FN_STR_POSITION: u32 = 0x00800000;
...
 pub const SQL_FN_TD_CURRENT_TIMESTAMP: u32 = 0x00080000;
 pub const SQL_FN_TD_EXTRACT: u32 = 0x00100000;
+pub const SQL_FN_TD_NOW: u32 = 0x00000001;
+#[allow(unused)]
+pub const SQL_FN_TD_CURDATE: u32 = 0x00000002;
+pub const SQL_FN_TD_DAYOFMONTH: u32 = 0x00000004;
+pub const SQL_FN_TD_DAYOFWEEK: u32 = 0x00000008;
+pub const SQL_FN_TD_DAYOFYEAR: u32 = 0x00000010;
+pub const SQL_FN_TD_MONTH: u32 = 0x00000020;
+pub const SQL_FN_TD_QUARTER: u32 = 0x00000040;
```
[the remark this answers is not in the document]
Comment (pmeredit) [reply odbc/src/api/definitions.rs:745]: We support quarter in timestamp_add and diff

**Review by nbagnard — APPROVED** [review]
LGTM!

File: core/src/bson_type_info.rs [hunk 12]
```diff
@@ -1,5 +1,7 @@
 use crate::definitions::SqlDataType;
 
+pub const STRING_SIZE: u16 = 65535;
```
Comment (terakilobyte) [inline core/src/bson_type_info.rs:3]:
```suggestion
pub const STRING_SIZE: u16 = u16::MAX;
```

File: odbc/src/api/col_attr_describe_tests.rs [hunk 13]
```diff
@@ -424,7 +425,11 @@ mod unit {
                         numeric_attr_ptr,
                     )
                 );
-                assert_eq!(expected, *numeric_attr_ptr);
+                let val = *numeric_attr_ptr;
+                assert_eq!(
+                    expected, val,
+                    "Expect value {expected} for {desc:?}, but got {val}"
```
Comment (terakilobyte) [inline odbc/src/api/col_attr_describe_tests.rs:431]: Love this

**Review by terakilobyte — APPROVED** [review]
LGTM. Just one small suggestion to fully remove the magic number.

**Review by nbagnard — CHANGES_REQUESTED** [review]

**Review by nbagnard — APPROVED** [review]
LGTM!

**Pull request closed without merging**